src/libecalc/dto/component_graph.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from typing import List
+
+from libecalc import dto
+from libecalc.common.component_info.component_level import ComponentLevel
+from libecalc.common.graph import Graph, NodeID
+from libecalc.dto.base import ComponentType
+from libecalc.dto.node_info import NodeInfo
+
+
+class ComponentGraph(Graph):
```

Comment:
Here I would want to use composition instead of inheritance, and make this "domain specific" wrt naming.